node/impl/full/eth.go

```diff
@@ -1270,7 +1271,33 @@ func (a *EthModule) EthCall(ctx context.Context, tx ethtypes.EthCall, blkParam e
 	return ethtypes.EthBytes{}, nil
 }
 
+func (e *EthEventHandler) GetEthLogsForBlockAndTransaction(ctx context.Context, blockHash *ethtypes.EthHash, txHash ethtypes.EthHash) ([]ethtypes.EthLog, error) {
+	ces, err := e.ethGetEventsForFilter(ctx, &ethtypes.EthFilterSpec{BlockHash: blockHash})
+	if err != nil {
+		return nil, err
+	}
+	logs, err := ethFilterLogsFromEvents(ctx, ces, e.SubManager.StateAPI)
+	if err != nil {
+		return nil, err
+	}
```

Comment:
```suggestion
	logs, err := e. EthGetLogs(ctx, &ethtypes.EthFilterSpec{BlockHash: blockHash})
	if err != nil {
		return nil, err
	}
```

isn't this the same thing? without needing to refactor `EthGetLogs`?

Reply:
@rvagg The return type of `EthGetLogs` i.e. `EthFilterResult` looks like:

```
type EthFilterResult struct {
	Results []interface{}
}
``` 

as it's supposed to be a generic type that  can work for the other ETH APIs as well that might simply send logs here. So we can't just use the output of `EthLogs` directly as we'll then have to do the messy wrangling of converting it to an `[]EthGetLogs`

Reply:
ohhh, that all makes sense now, sorry, should have spotted that